.circleci/config.yml

```diff
@@ -307,53 +307,88 @@ jobs:
             kill $TEKU_PID
             exit $EXIT_CODE
       - store_artifacts:
-          path: openapi.json
+          path: ./docs/openapi.json
       - store_artifacts:
           path: teku_output.log
       - persist_to_workspace:
           root: ~/project
           paths:
-            - ./
+            - ./docs/openapi.json
 
   publishAPIDoc:
     executor: node_executor
     steps:
+      # This job requires that we checkout current branch first to have all the repos.
+      # it will not be used directly as we have the saved workspace
+      # but will enable to checkout gh-pages branch.
       - checkout
       - attach_workspace:
-          at: ~/project
-      - add_ssh_keys:
-          fingerprints:
-            - '45:36:58:67:43:15:c6:5f:2c:58:ec:7f:71:e2:e6:ef'
+          at: /tmp/workspace
       - run:
-          name: Set Git user params
+          name: checkout gh pages branch and copy files from workspace
           command: |
-            git config --global user.name $CIRCLE_USERNAME
-            git config --global user.email "${CIRCLE_USERNAME}@users.noreply.github.com"
+            # publishing happens by pushing files to gh-pages branch
+            # retrieve branch to publish to from CIrcle CI env vars and fallback on test-gh-pages
+            # if no var defined
+            [ -n "$BRANCH_OVERRIDE" ] && readonly BRANCH="$BRANCH_OVERRIDE" || readonly BRANCH='test-gh-pages'
```

Comment:
I'd suggest we should fall back to `gh-pages` so that the code says what it will do in the normal case.  Removing the `BRANCH_OVERRIDE` is probably ideal if we can so that you don't have to go digging through CircleCI to find out what this will actually do.

Reply:
This code was suggested by @benjamincburns so I would like his opinion on that.

Reply:
I agree with @ajsutton - his approach is better than the one I suggested.